### PR TITLE
LPS-112044 To add Site Templates to dynamic-data-mapping module

### PR DESCRIPTION
**Pull request by david-gutierrez-mesa**
https://issues.liferay.com/browse/LPS-112044

### CODE REVIEW

**Comment by david-gutierrez-mesa** [comment]
ci:test:sf

**Comment by david-gutierrez-mesa** [comment]
ci:test:sf

**Comment by jpince** [comment]
@david-gutierrez-mesa my suggestions:
1. I think you should break this up into smaller pulls so you can actually get these changes in.
1. Figure out how to make changes in the other components and make sure those test leads are aware. Especially for themes, since FI has been _removing_ tests from acceptance
1. This is not correct, you are setting a property twice https://github.com/liferay-echo/liferay-portal/pull/286/commits/2fe9de6ec0b0fa24a6a8072ee4857df50b2ed5b5#diff-676028f4fc5d47e0f20c67e2ebeadeb5R7. Use `testray.component.names` instead. @kenjiheigel should POSHI validation catch this kind of error where a property is set twice? Not sure how common/useful this would be.

cc/ @manoelcyreno

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsUI.testcase [hunk 1]
```diff
@@ -4,6 +4,7 @@ definition {
 	property portal.release = "true";
 	property portal.upstream = "true";
 	property testray.main.component.name = "Fragments";
+	property testray.main.component.name = "Theme";
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsUI.testcase:7]: remove this line, please ...

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithFragmentConfiguration.testcase [hunk 2]
```diff
@@ -4,6 +4,7 @@ definition {
 	property portal.release = "true";
 	property portal.upstream = "true";
 	property testray.main.component.name = "Fragments";
+	property testray.main.component.name = "Theme";
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithFragmentConfiguration.testcase:7]: remove this line, please ...

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesWithFragmentConfiguration.testcase [hunk 3]
```diff
@@ -4,6 +4,7 @@ definition {
 	property portal.release = "true";
 	property portal.upstream = "true";
 	property testray.main.component.name = "Fragments";
+	property testray.main.component.name = "Theme";
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesWithFragmentConfiguration.testcase:7]: remove this line, please ...

**Comment by john-co** [comment]
@david-gutierrez-mesa and I discussed this briefly Friday through chat. I suggested using `property testray.component.names = "Theme";` so probably a typo. I suggest just remove `main.`.

**Comment by kenjiheigel** [comment]
@jpince, it's grey whether it should or not.. but I don't think we've encountered this before, so it's never come up for question to my knowledge. 

If you guys want that to be validated, create a poshi ticket for it and let me know, but it wouldn't be high priority cause it doesn't look like it's breaking anything. @david-gutierrez-mesa , @manoelcyreno

**Comment by david-gutierrez-mesa** [comment]
@jpince thank you for the review. About your points:
1. I don't get this one. I grouped in order to don't waste resources executing ci:forward for just changes in the tests to be executed
2. I contacted @john-co and @alexcandidos to get their approval. I was not going to send changes to BChan if they didn't agree. I even asked John to review the PR and do the ci:forward if he was ok with changes
3. It was misunderstood. As John said, we had a conversation and he suggested adding two components. I understood we could just add them as main and I didn't realize one must be main and the other can't. I'm fixing and submitting it. My bad. Anyway, John was going to review it.

Let us know if we could go ahead with this or we need to change something else or any extra review.

**Comment by david-gutierrez-mesa** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsUI.testcase (hunk 1 above)
Comment (david-gutierrez-mesa) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsUI.testcase]: removed main

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesWithFragmentConfiguration.testcase (hunk 3 above)
Comment (david-gutierrez-mesa) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesWithFragmentConfiguration.testcase]: removed main

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithFragmentConfiguration.testcase (hunk 2 above)
Comment (david-gutierrez-mesa) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithFragmentConfiguration.testcase]: removed main

**Comment by jpince** [comment]
@david-gutierrez-mesa as long as they're aware, then this is fine, thanks.

**Comment by david-gutierrez-mesa** [comment]
ci:forward

**Comment by david-gutierrez-mesa** [comment]
ci:test:stable

**Comment by david-gutierrez-mesa** [comment]
ci:test:stable

**Comment by david-gutierrez-mesa** [comment]
ci:forward

**Comment by david-gutierrez-mesa** [comment]
ci:test:relevant

**Comment by david-gutierrez-mesa** [comment]
ci:forward

**Pull request closed without merging**